### PR TITLE
simplify `describe_field` func in borrowck's diagnostics part

### PR DESCRIPTION
**Pull request by SparrowLii**
This PR simplify the `describe_field` func in borrowck's diagnostics part, besides fix the FIXME in it.

### CODE REVIEW

**Review by jackh726 — CHANGES_REQUESTED** [review]
One small thing

File: compiler/rustc_borrowck/src/diagnostics/mod.rs [hunk 1]
```diff
@@ -329,30 +330,20 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
 
     /// End-user visible description of the `field`nth field of `base`
     fn describe_field(&self, place: PlaceRef<'tcx>, field: Field) -> String {
-        // FIXME Place2 Make this work iteratively
-        match place {
-            PlaceRef { local, projection: [] } => {
-                let local = &self.body.local_decls[local];
-                self.describe_field_from_ty(local.ty, field, None)
-            }
+        let place_ty = match place {
```
[the remark this answers is not in the document]
Comment (SparrowLii) [reply compiler/rustc_borrowck/src/diagnostics/mod.rs:333]: Yea we need to know the value of `variant_index`, and the structure of `PlaceTy` fits well. But it also makes sense to separate `Ty` and `variant_index` for semantic clarity

[the remark this answers is not in the document]
Comment (jackh726) [reply compiler/rustc_borrowck/src/diagnostics/mod.rs:333]: Whoops I missed the variant index.

**Pull request merged**